blocks/graph.py

```diff
@@ -354,3 +387,64 @@ def apply_noise(computation_graph, variables, level, seed=None):
         replace[variable] = (variable +
                              rng.normal(variable.shape, std=level))
     return computation_graph.replace(replace)
+
+
+def apply_dropout(computation_graph, variables, drop_prob=0.5, rng=None,
+                  seed=None):
+    """Returns a graph to variables in a computational graph.
+
+    Parameters
+    ----------
+    computation_graph : instance of :class:`ComputationGraph`
+        The computation graph.
+    variables : list of :class:`~tensor.TensorVariable`
+        Variables to be dropped out.
+    drop_prob : float
+        Probability of dropping out. If you want to apply the dropout
+        with different probabilities for different layers, call it
+        several times.
+    rng : :class:`~theano.sandbox.rng_mrg.MRG_RandomStreams`
+        Random number generator.
+    seed : int
+        Random seed to be used if `rng` was not specified.
+
+    Notes
+    -----
+    For more information, see [DROPOUT]_.
+
+    .. [DROPOUT] Hinton et al. *Improving neural networks by preventing
+       co-adaptation of feature detectors*, arXiv:1207.0580.
+
+    Examples
+    --------
+    >>> from theano import tensor
```

Comment:
This doctest is great stuff, though it might be good to interleave a few `>>> # Comments` to help the exposition of what you're demonstrating.